pkg/apis/nmstate/v1alpha1/nodenetworkconfigurationpolicy_types.go

```diff
@@ -30,36 +48,65 @@ type NodeNetworkConfigurationPolicySpec struct {
 // NodeNetworkConfigurationPolicyStatus defines the observed state of NodeNetworkConfigurationPolicy
 // +k8s:openapi-gen=true
 type NodeNetworkConfigurationPolicyStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
-	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
+	Enactments EnactmentList `json:"enactments,omitempty" optional:"true"`
 }
 
-// +genclient
-// +genclient:nonNamespaced
-// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// TODO: This is a temporary solution. This list will be replaced by a dedicated
+// NodeNetworkConfigurationEnactment object.
+type EnactmentList []Enactment
 
-// NodeNetworkConfigurationPolicy is the Schema for the nodenetworkconfigurationpolicies API
-// +k8s:openapi-gen=true
-// +kubebuilder:subresource:status
-// +kubebuilder:resource:path=nodenetworkconfigurationpolicies,shortName=nncp
-type NodeNetworkConfigurationPolicy struct {
-	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
+type Enactment struct {
```

Comment:
NodeNetworkConfigurationEnactment

Reply:
NodeNetworkConfigurationEnactment will be the name of the new object. It will be a full object with Metadata, State etc... I don't want it to be confused with this list. The NodeNetworkCofigurationPolicy will have a list of all Enactments (having key .nodeNetworkConfigurationEnactment is a little too much). This list will contain link to enactment or simply matched=False and name of respective node.

This list will remain here, just its contents will be moved farther to NodeNetworkConfigurationEnactment obejcts. Do we want to gave `policy.enactments` or `policy.nodes`? I'm kinda in favor of the former.

Reply:
ok, fair enough, I am for policy.enactments.